### PR TITLE
Rewrite build script as a Makefile

### LINKED ISSUE #130: Use Makefile instead of script for building
By replacing parts of the build script with simple `make`  targets we can make the build process more modular, easier to parallelize and to debug.

### PR DESCRIPTION
**Pull request by pviotti**
- make the build more modular (single targets can be build without touching the script, e.g. `make clone-all`)
 - speed up builds (`make -jX` where `X` is for instance the number of core of your machine, enables parallel builds of targets)
    - update: the parallelization flag *may* cause issues since the dump and compilation targets both use the docker-compose db commands, which may result in race conditions and errors

I've only tested the Linux64 distribution created with the Makefile.

Close #130

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,3 +30,30 @@ oh-admin-manual.pdf:
...
+dw-jre-all: jre-linux32.tar.gz jre-linux64.tar.gz jre-win.zip
+dw-mysql-all: mysql-linux32.tar.gz mysql-linux64.tar.gz mysql-win.zip
+jre-linux32.tar.gz:
+	wget -q -nc https://cdn.azul.com/zulu/bin/zulu11.39.15-ca-jre11.0.7-linux_i686.tar.gz -O jre-linux32.tar.gz
+jre-linux64.tar.gz:
+	wget -q -nc https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.7%2B10/OpenJDK11U-jre_x64_linux_hotspot_11.0.7_10.tar.gz -O jre-linux64.tar.gz
+jre-win.zip:
+	wget -q -nc https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.7%2B10.2/OpenJDK11U-jre_x86-32_windows_hotspot_11.0.7_10.zip -O jre-win.zip
+mysql-linux32.tar.gz:
+	wget -q -nc https://downloads.mysql.com/archives/get/p/23/file/mysql-5.7.30-linux-glibc2.12-i686.tar.gz -O mysql-linux32.tar.gz
```
Comment (elfosardo) [inline Makefile]: just a suggestion that maybe using a variable to define versions or common path here and for jdk above would simplify maintaining the code

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,144 @@
...
+.PHONY: clone-all clean clean-downloads dw-all dw-jre-all dw-mysql-all compile-all docs-all
 
-build:
-	./build_poh.sh
+all: compile-all dw-all release-files
 
-clean: clean-downloads
+# Clean targets
+clean:
 	git clean -xdff
```
Comment (mwithi) [inline Makefile:40]: I would add something to avoid IDE files removal (like .project for Eclipse, .vscode/ for VScode, etc...)

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,144 @@
...
+FULL := openhospital-$(OH_VERSION)
+WIN := poh-win32-$(POH_VERSION)-core-$(OH_VERSION)
+LINUX32 := poh-linux-x32-$(POH_VERSION)-core-$(OH_VERSION)
+LINUX64 := poh-linux-x64-$(POH_VERSION)-core-$(OH_VERSION)
+JRE_WIN := jre-win.zip
+JRE_LINUX32 := jre-linux32.tar.gz
+JRE_LINUX64 := jre-linux64.tar.gz
+MYSQL_WIN := mysql-win.zip
+MYSQL_LINUX32 := mysql-linux32.tar.gz
+MYSQL_LINUX64 := mysql-linux64.tar.gz
```
Comment (mwithi) [inline Makefile:25]: I would add some comments for the possible usages

File: Makefile [hunk 4]
```diff
@@ -1,11 +1,144 @@
+SHELL := /bin/bash
+.SHELLFLAGS := -eu -o pipefail -c
+.ONESHELL: 			# single recipes are run in one bash session, instead of one per line
+.DELETE_ON_ERROR:		# delete the target if its recipe failed
+OH_VERSION ?= $(shell git describe --abbrev=0 --tags)
+POH_VERSION ?= 1.0
```
Comment (mwithi) [inline Makefile]: last distributed POH_VERSION version was 0.0.6

File: Makefile [hunk 5]
```diff
@@ -25,7 +25,9 @@ help:
 	@echo -e "\t$(JRE_WIN), $(JRE_LINUX32), $(JRE_LINUX64), $(MYSQL_WIN), $(MYSQL_LINUX32), $(MYSQL_LINUX64)"
 
 # Clean targets
-clean:
+clean: clean-downloads
+	rm -rf release-files core gui doc poh-linux* poh-win* CHANGELOG CHANGELOG.md database.sql *.pdf
+clean-all:
 	git clean -xdff
```
Comment (mwithi) [inline Makefile]: just proposal: instead  to proceed with the blacklist could we approach with git clean exclusion (that should be shorter)?

`git clean -xdff -e ".project" -e ".vscode"` (and adding more when needed)

**Review by mwithi — APPROVED** [review]
A part the proposal on the clean targets, I think is very ok

File: Makefile (hunk 4 above)
Comment (pviotti) [reply Makefile]: yes. The POH has been around for many years and we have recently upgrade both db and java, so I think a version bump is due - maybe 0.1? Can be done in another PR anyway.

Comment (mwithi) [reply Makefile]: yeah it's true, today I was thinking the same (the comment is from yesterday) and we should have done it in #121, #129 and #134... we can updated it now to 0.1 (nobody will complain) and then we change to 1.0 when POH will "include" optional JVM and File-based DB 👍

**Pull request merged**